lnd.go

```diff
@@ -1140,3 +1115,91 @@ func waitForWalletPassword(cfg *Config, restEndpoints []net.Addr,
 		return nil, fmt.Errorf("shutting down")
 	}
 }
+
+// initializeDatabases extracts the current databases that we'll use for normal
+// operation in the daemon. Two databases are returned: one remote and one
+// local. However, only if the replicated database is active will the remote
+// database point to a unique database. Otherwise, the local and remote DB will
+// both point to the same local database.
+func initializeDatabases(ctx context.Context, cfg *Config) (*channeldb.DB, *channeldb.DB, error) {
+	ltndLog.Infof("Opening the main database, this might take a few " +
+		"minutes...")
+
+	if cfg.DB.Backend == lncfg.BoltBackend {
+		ltndLog.Infof("Opening bbolt database, sync_freelist=%v",
+			cfg.DB.Bolt.SyncFreelist)
+	}
+
+	startOpenTime := time.Now()
+
+	databaseBackends, err := cfg.DB.GetBackends(
+		ctx, cfg.localDatabaseDir(), cfg.networkName(),
+	)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to obtain database "+
+			"backends: %v", err)
+	}
+
+	// If the remoteDB is nil, then we'll just open a local DB as normal,
+	// having the remote and local pointer be the exact same instance.
+	var localChanDB, remoteChanDB *channeldb.DB
+	if databaseBackends.RemoteDB == nil {
+		// Open the channeldb, which is dedicated to storing channel,
+		// and network related metadata.
+		localChanDB, err = channeldb.CreateWithBackend(
+			databaseBackends.LocalDB,
+			channeldb.OptionSetRejectCacheSize(cfg.Caches.RejectCacheSize),
+			channeldb.OptionSetChannelCacheSize(cfg.Caches.ChannelCacheSize),
+			channeldb.OptionDryRunMigration(cfg.DryRunMigration),
+		)
+		if err != nil {
+			err := fmt.Errorf("unable to open local channeldb: %v", err)
+			ltndLog.Error(err)
```

Comment:
this log shouldn't happen if the dry run succeeded. i think what we want is
```golang
switch {
case err == channeldb.ErrDryRunMigrationOK:
    return nil, nil, err

case err != nil:
    err := fmt.Errorf("unable to open local channeldb: %v", err)
    ltndLog.Error(err)
    return nil, nil, err
}
```
since the successful dry run log is still done at the top level (also see my other comment)

Reply:
If the dry run succeeds (for the local DB), then as written, this'll then go and also run it for the remote db. There's an assumption here that we want to make sure the dry run succeeds on both the local and remote DB. Modified the section for the remote DB to only return early and not also log.

Reply:
This is for the local-only case. Currenlty it will log "unable to open local channeldb", return `ErrDryRunMIgrationOK` then log the success log at the call site.

Reply:
Latest version should be correct now, pushed up the intended new diff along w/ a slight change.